### PR TITLE
438-Send an Email along with creating the userInvite row

### LINKED ISSUE #438: Create sendUserInvite resolver that sends the email after creating the user invite row
## Summary:
We want to create a new resolver (or update it if it already exists from #438) which it creates a user invite record in our database, and sends an invite link to the invited email at `https://sistering-dev.web.app/create-account?token={new user invite row record id}`. This makes it so we can securely generate and send out user invites to users from our admin platform.

## Technical:
- Create a `sendUserInvite` resolver in the `userResolver` or just update it to send an stub email with our email service `sendEmail` helper
- After creating the user invite record in #438 in the resolver, we want to send an email to the invited user with a link to https://sistering-dev.web.app/create-account?token={new user invite record id}
- For this ticket, just ensure to handle sending out emails
- The stub email should just contain the text:
```html
Hello,

You have received an user invite to join the Sistering volunteer platform. Please click the following link to reset it. This link is only valid for 2 weeks.

Create Account<link attached: https://sistering-dev.web.app/create-account?token={new user invite record id}>
```

### PR DESCRIPTION
**Pull request by eevanwong**
## Ticket link

Closes #438 

## Implementation description
* The createUserInvite resolver also sends an email once the userInvite is inserted into the database.

## Steps to test
1. Use this query:
```
mutation testCreateUserInvite($email: String!) {
   createUserInvite(email: $email, role: VOLUNTEER) {
     uuid
   }
}
```
And in your query variables, use whichever email you want (spam or smthn) and run the mutation, it should send this email:
image

The link will take you to the page to create your account:
image

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/graphql/index.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ const isValidDateTime = (dateTimeString: string) => {
   return (
     !Number.isNaN(Date.parse(`${dateTimeString}:00`)) && // cover cases of DD > 31
     new Date(`${dateTimeString}:00+00:00`).toISOString().slice(0, 16) ===
-      dateTimeString
+    dateTimeString
```
Comment (MatoPlus) [inline backend/graphql/index.ts:54]: huh, is our column length limit this small?

File: backend/graphql/index.ts [hunk 2]
```diff
@@ -167,7 +167,7 @@ const graphQLMiddlewares = {
     updateUser: authorizedByAdmin(),
     deleteUserById: authorizedByAdmin(),
     deleteUserByEmail: authorizedByAdmin(),
-    createUserInvite: authorizedByAdmin(),
+    createUserInvite: authorizedByAllRoles(),
```
Comment (MatoPlus) [inline backend/graphql/index.ts:170]: Hmm can you explain this change? Does doesn't look right. Only admins should be able to invite other admins + employees

File: backend/graphql/index.ts (hunk 1 above)
Comment (MatoPlus) [reply backend/graphql/index.ts:54]: I don't think my local eslint likes this change but no action needed

File: backend/services/implementations/userService.ts [hunk 3]
```diff
@@ -465,14 +467,28 @@ class UserService implements IUserService {
...
         email: userInvite.email,
         role: userInvite.role.toString() as Role,
         uuid: userInvite.uuid,
       };
     } catch (error: unknown) {
       Logger.error(
-        `Failed to create user invite row. Reason = ${getErrorMessage(error)}`,
+        `Issue with either creating user invite row or sending email. Reason = ${getErrorMessage(
+          error,
+        )}`,
```
Comment (MatoPlus) [inline backend/services/implementations/userService.ts:491]: idem, this can be reverted

File: backend/services/interfaces/userService.ts [hunk 4]
```diff
@@ -100,7 +101,11 @@ interface IUserService {
    * @param role role that the user will have
    * @throws Error if invite creation fails
    */
-  createUserInvite(email: string, role: Role): Promise<UserInviteResponse>;
+  createUserInvite(
+    email: string,
+    role: Role,
+    emailService: IEmailService,
+  ): Promise<UserInviteResponse>;
```
Comment (MatoPlus) [inline backend/services/interfaces/userService.ts:108]: idem

File: backend/graphql/resolvers/userResolvers.ts [hunk 5]
```diff
@@ -111,7 +111,7 @@ const userResolvers = {
       _parent: undefined,
       { email, role }: { email: string; role: Role },
     ): Promise<UserInviteResponse> => {
-      return userService.createUserInvite(email, role);
+      return userService.createUserInvite(email, role, emailService);
```
Comment (MatoPlus) [inline backend/graphql/resolvers/userResolvers.ts:114]: idem

**Review by MatoPlus — CHANGES_REQUESTED** [review]
Looks great overall! Let's clean the code a bit and ship it 🧹

File: backend/graphql/index.ts (hunk 2 above)
Comment (eevanwong) [reply backend/graphql/index.ts:170]: Thats my bad, I think I made that change to test something when I was dealing with a bug but forgot to revert it.

File: backend/graphql/index.ts (hunk 1 above)
Comment (MatoPlus) [reply backend/graphql/index.ts:54]: oh here's the same linting issue that appeared on my local machine

File: backend/services/implementations/userService.ts [hunk 6]
```diff
@@ -472,7 +473,9 @@ class UserService implements IUserService {
       };
     } catch (error: unknown) {
       Logger.error(
-        `Failed to create user invite row. Reason = ${getErrorMessage(error)}`,
+        `Issue with either creating user invite row. Reason = ${getErrorMessage(
+          error,
+        )}`,
```
Comment (MatoPlus) [inline backend/services/implementations/userService.ts:478]:
```suggestion
        `Failed to create user invite row. Reason = ${getErrorMessage(error)}`,
```
Let's revert

**Review by MatoPlus — COMMENTED** [review]
There's still some unused imports. Besides the linting problems, code looks good to merge!

**Review by MatoPlus — APPROVED** [review]
Ship it!

File: backend/graphql/resolvers/userResolvers.ts [hunk 7]
```diff
@@ -111,7 +111,17 @@ const userResolvers = {
...
-      return userService.createUserInvite(email, role);
+      const results = await userService.createUserInvite(email, role);
+      const SUBJECT = "Invitation to Sistering Platform";
+      const htmlBody = `
+<h3>Hello,</h3>
+<br/>
+<h3>You have received a user invite to join the Sistering volunteer platform. Please click the following link to set up your account. This link is only valid for 2 weeks.<h3/>
+<br/>
+<a href="https://sistering-dev.web.app/create-account?token=${results.uuid}>">Create Account</a>
+`;
```
Comment (MatoPlus) [inline backend/graphql/resolvers/userResolvers.ts:122]: @eevanwong can you use the new template I provided? Please rebase and use that instead.

**Review by MatoPlus — CHANGES_REQUESTED** [review]

File: backend/graphql/resolvers/userResolvers.ts (hunk 7 above)
Comment (MatoPlus) [reply backend/graphql/resolvers/userResolvers.ts:122]: https://github.com/uwblueprint/sistering/pull/490/files#diff-44e16111f2add2306d8f7075373015dde35857eb70c1e351491e7e7452f60942R35

File: backend/graphql/resolvers/userResolvers.ts [hunk 8]
```diff
@@ -111,7 +112,15 @@ const userResolvers = {
       _parent: undefined,
       { email, role }: { email: string; role: Role },
     ): Promise<UserInviteResponse> => {
-      return userService.createUserInvite(email, role);
+      const results = await userService.createUserInvite(email, role);
+      const SUBJECT = "Invitation to Sistering Platform";
```
Comment (MatoPlus) [inline backend/graphql/resolvers/userResolvers.ts:116]: let's update this to match design I think it should be "Welcome to your ${role} account"

Comment (MatoPlus) [reply backend/graphql/resolvers/userResolvers.ts:116]: https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=1643%3A48828

File: backend/graphql/resolvers/userResolvers.ts [hunk 9]
```diff
@@ -111,7 +112,15 @@ const userResolvers = {
...
+
+      const htmlBody = accountCreationInviteTemplate(
+        results.role,
```
Comment (MatoPlus) [inline backend/graphql/resolvers/userResolvers.ts:119]: this should be title case, eg: `Volunteer` or `Administrator`

**Review by MatoPlus — APPROVED** [review]

**Pull request merged**